core/src/main/java/discord4j/core/event/dispatch/GuildDispatchHandlers.java

```diff
@@ -163,6 +163,22 @@
                 .and(startMemberChunk)
                 .thenReturn(new GuildCreateEvent(gateway, context.getShardInfo(), new Guild(gateway, guild)));
     }
+    
+    private static VoiceStateData wrapVoiceState(VoiceStateData voiceState, String guildId) {
+    	return ImmutableVoiceStateData.builder()
+    			.guildId(Possible.of(guildId))
+    			.channelId(voiceState.channelId())
+    			.userId(voiceState.userId())
+    			.member(voiceState.member())
+    			.sessionId(voiceState.sessionId())
+    			.deaf(voiceState.deaf())
+    			.mute(voiceState.mute())
+    			.selfDeaf(voiceState.selfDeaf())
+    			.selfMute(voiceState.selfMute())
+    			.selfStream(voiceState.selfStream())
+    			.suppress(voiceState.suppress())
```

Comment:
Here you can use `.from(voiceState)` instead of copying all fields one by one, it saves a lot of lines of code and is less prone to oversights.

Overall I'd honestly say it's not bad for a first PR :+1: I'll let the D4J team give further review.